packages/debug/src/node/debug-adapter-factory.ts

```diff
@@ -67,10 +69,12 @@ export class LaunchBasedDebugAdapterFactory implements DebugAdapterFactory {
             !!forkOptions && !!forkOptions.modulePath;
 
         const processOptions: RawProcessOptions | RawForkOptions = { ...executable };
-        const options = { stdio: ['pipe', 'pipe', 2] };
+        const options: { stdio: (string | number)[], env?: object, execArgv?: string[] } = { stdio: ['pipe', 'pipe', 2] };
 
         if (isForkOptions(processOptions)) {
             options.stdio.push('ipc');
+            options.env = environment.electron.runAsNodeEnv();
+            options.execArgv = (executable as DebugAdapterForkExecutable).execArgv;
```

Comment:
is not it redundant? `executable` is already spread into `processOptions` here: https://github.com/theia-ide/theia/pull/5508/files#diff-0edd7bfb7a3a6822ad4c2bf344b5f8f1R71